### PR TITLE
minor book improvements

### PR DESCRIPTION
**Pull request by kaelari**

### CODE REVIEW

File: patch/core/book.js [hunk 1]
```diff
@@ -26,19 +23,20 @@ module.exports = {
 			} else if (item.includes('crystal')) {
 				item = 'crystal';
 			}
-
+
```
Comment (BloodSweatAndCode) [inline patch/core/book.js]: Remove this unnecessary whitespace. I typically set the "remove trailing whitespace" preference in my editor's settings, not sure which one you're using.

File: patch/core/book.js [hunk 2]
```diff
@@ -26,19 +23,20 @@ module.exports = {
...
 			}
-
+			
 			// set clue text based on actor type
 			if (actor === 'Death') {
 				clues.push('Death guards\n' + item);
 			} else if (actor === 'Camilla') {
 				clues.push('Camilla\ndefends\n' + item);
 			} else if (actor === 'merchant') {
-				clues.push(item + '\nfor sale in\n' + location);
```
Comment (BloodSweatAndCode) [inline patch/core/book.js]: I see the comment here, but I don't fully understand why basic merchants are being excluded. Knowing any given item is for sale in a town, particularly if it's a holy-water blocked one, could significantly affect routing and make the book _really_ worth grabbing.

File: patch/core/book.js [hunk 3]
```diff
@@ -26,19 +23,20 @@ module.exports = {
...
+				//ignore basic merchants to make clues more relevant
+				//clues.push(item + '\nfor sale in\n' + location);
 			} else if (actor === 'sacred flame') {
 				clues.push(item + '\nis hidden\non Dabi\'s Path');
 			} else if (actor === 'orb') {
 				clues.push(item + '\nhidden in\n' + location + ' orb');
-			} else if (actor === 'crystal dude') {
+			} else if (actor === 'crystal dude' || actor === 'secret merchant') {
```
Comment (BloodSweatAndCode) [inline patch/core/book.js]: Given that secret merchants require garlic, they aren't technically free. It's worth it (and fun) to note the necessity of garlic in a separate text for secret merchants

File: patch/core/book.js [hunk 4]
```diff
@@ -52,8 +50,20 @@ module.exports = {
 				const maxlength = a.text.length;
 
 				// make sure new text does not exceed the text it is replacing
-				while (clues[index].length > maxlength) {
```
Comment (BloodSweatAndCode) [inline patch/core/book.js]: good catch on the `while` here, no reason for it

File: patch/core/book.js [hunk 5]
```diff
@@ -52,8 +50,20 @@ module.exports = {
...
-					clues[index] = clues[index].slice(0, maxlength - 1);
+				if (clues[index].length > maxlength) {
+					//trim some words out to try to make it fit
+					clues[index] = clues[index].replace(/ ?mansion/i, '');
+					clues[index] = clues[index].replace(/ ?graveyard/i, '');
+					clues[index] = clues[index].replace(/ ?woods/i, '');
+					clues[index] = clues[index].replace(/ ?cemetery/i, '');
```
Comment (BloodSweatAndCode) [inline patch/core/book.js]: I like these edits, but I think "cemetery" could cause confusing messages. Changing "Camilla cemetery" to "Camilla" would inevitably lead many players to think that Camilla is holding the item, not that they need garlic to get an item in Camilla cemetery. I would suggest just some custom handling of that one condition, probably up in the `location.replace()` normalization at the top of the file to handle "Camilla cemetery" and just replace it with "cemetery". That, combined with giving garlic secret merchants their own text (as mentioned in a prior review comment) should make the clue much more clear.

File: patch/core/book.js (hunk 2 above)
Comment (BloodSweatAndCode) [reply patch/core/book.js]: Also, comments should have a space between the `//` and the text of the comment

File: patch/core/book.js (hunk 5 above)
Comment (BloodSweatAndCode) [reply patch/core/book.js]: Also, that series of regexps (after excluding "cemetery" with my aforementioned suggestion) could be streamlined as:

```
clues[index] = clues[index].replace(/ ?(?:mansion|graveyard|woods)/i, '');
```

`(?:)` indicates a "non-capturing group", so it allows you to match on multiple values with getting into all the subgroup matching regexps can do.

File: patch/core/book.js (hunk 2 above)
Comment (kaelari) [reply patch/core/book.js]: in playing i found getting a merchant notice to always be annoying. you have to go out of your way to get a book and i wanted them to have important(-ish) information, i care about the bosses and orbs the most, merchants were just bottom of the list and never seemed to matter. maybe a check so only merchants behind holy water walls would be better? going out of my way to find a merchant in jova has a whip is just bleck... your call of course but i'd rather not have merchants in the list

File: patch/core/book.js (hunk 3 above)
Comment (kaelari) [reply patch/core/book.js]: couldn't think of anything good. how would you phrase it?

File: patch/core/book.js (hunk 5 above)
Comment (kaelari) [reply patch/core/book.js]: using the replace anyway it's easy enough to put 
clues[index] = clues[index].replace(/camilla cemetery/i, 'cemetery');
I don't see an advantage to putting them all in one line, but doesn't really matter. If thats how you want it.

**Comment by BloodSweatAndCode** [comment]
resolved in #95
